docs/decisions/0017-reimplement-asset-processing.rst

```diff
@@ -217,61 +216,55 @@ The three top-level edx-platform asset processing actions are *build*, *collect*
 
        Paver task that invokes ``webpack --watch`` for Webpack assets and watchdog (a Python library) for other assets.
 
-     - ``scripts/build-assets.sh --watch <stage>``
-
-       (where ``<stage>`` is optionally one of the build stages described above. If provided, only that stage's assets will be watched.)
+     - ``npm run watch``
 
        Bash wrappers around invocation(s) of `watchman <https://facebook.github.io/watchman/>`_, a popular file-watching library maintained by Meta. Watchman is already installed into edx-platform (and other services) via the pywatchman pip wrapper package.
 
-       Note: This adds a Python dependency to build-assets.sh. However, we could be clear that watchman is an *optional* dependency of build-assets.sh which enables the optional ``--watch`` feature. This would keep the *build* action Python-free. Alternatively, watchman is also available Python-free via apt and homebrew.
 
 Build Configuration
 -------------------
 
-``scripts/build-assets.sh`` will accept various command-line options to configure the build. It will also accept the same options in the form of the ``OPENEDX_BUILD_ASSETS_OPTS`` enviroment variable. Options from the environment variable will be processed first, and then overridden by options provided on the command line. The environment variable allows distributions like Tutor to seed the build script with "defaults" in the event that the upstream defaults are not sufficient, while still allowing individual operators to run the script with whichever options they like.
+To facilitate a generally Python-free build reimplementation, we will require that certain Django settings now be specified as environment variables, which can be passed to the build like so::
 
-As a concrete example, the default value of ``--theme-dirs`` will be ``''`` (that is: no themes) and the default value of ``--static-root`` will be ``./test_root/static``. Neither of those are suitable for Tutor. Instead, Tutor will set the environment variable in its Dockerfile::
+  MY_ENV_VAR="my value" npm run build    # Set for the whole build.
+  MY_ENV_VAR="my value" npm run webpack  # Set for just a single step, like webpack.
 
-  ...
-  ENV OPENEDX_BUILD_ASSETS_OPTS '--theme-dirs /openedx/themes --static-root /openedx/staticfiles'
-  ...
+For Docker-based distributions like Tutor, these environment variables can instead be set in the Dockerfile.
 
-Later, in the container, a user might run::
```

Comment:
This whole `OPENEDX_BUILDS_ASSETS_OPTS` idea doesn't play well with the simpler `npm run` command structure, so I removed it in favor of simple environment variables (below).